File: neetcode/Array/Group_Anagrams/GroupAnagrams.java
// https://leetcode.com/problems/group-anagrams
// https://neetcode.io/problems/anagram-groups?list=blind75

class GroupAnagrams {

  public List<List<String>> groupAnagrams(String[] strs) {
    HashMap<String,List<String>> map = new HashMap<>();

    for(String str : strs) {
      char[] ch = str.toCharArray();
      Arrays.sort(ch);
      String sortedWord = new String(ch);
      if(!map.containsKey(sortedWord))
        map.put(sortedWord, new ArrayList<>());
      map.get(sortedWord).add(str);
    }
    return new ArrayList<>(map.values());
  }
}

File: neetcode/python/Array/group_anagrams.py
# https://leetcode.com/problems/group-anagrams
# https://neetcode.io/problems/anagram-groups?list=blind75

class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        str_map = {}
        for string in strs:
            sorted_string = "".join(sorted(string))
            if sorted_string in str_map:
                str_map[sorted_string].append(string)
            else:
                str_map[sorted_string] = [string]

        return list(str_map.values())
